lib/core/utilities.js

```diff
@@ -110,3 +112,116 @@ export function getProp (holder, propName) {
 
   return result
 }
+
+const TokenExpirationStatusEnum = Object.freeze({
+  UNKNOWN: 'UNKNOWN',
+  VALID: 'VALID',
+  EXPIRED: 'EXPIRED',
+  REFRESH_EXPIRED: 'REFRESH_EXPIRED'
+})
+
+export class TokenExpirationStatus {
+  constructor (scheme) {
+    this.$scheme = scheme
+  }
+
+  _calculateTokenStatus () {
+    const now = Date.now()
+    let tokenExpiresAt, refreshTokenExpiresAt
+
+    try {
+      tokenExpiresAt = this.$scheme._getTokenExpiration()
+      refreshTokenExpiresAt = this.$scheme._getRefreshTokenExpiration()
+
+      if (tokenExpiresAt === false) {
+        return TokenExpirationStatusEnum.UNKNOWN
+      }
+    } catch (error) {
+      return TokenExpirationStatusEnum.UNKNOWN
+    }
+
+    // Give us some slack to help the token from expiring between validation and usage
+    const timeSlackMillis = 500
+    tokenExpiresAt -= timeSlackMillis
+    refreshTokenExpiresAt -= timeSlackMillis
+
+    // Token is still valid
+    if (now < tokenExpiresAt) {
+      return TokenExpirationStatusEnum.VALID
+    }
+
+    if (now > refreshTokenExpiresAt) {
+      return TokenExpirationStatusEnum.REFRESH_EXPIRED
+    }
+
+    return TokenExpirationStatusEnum.EXPIRED
+  }
+
+  setTokenExpiration (result = {}) {
+    const token = this.$scheme.$auth.getToken(this.$scheme.name)
+    let decodedToken, tokenExpiration
+    const _tokenIssuedAt = getProp(result, this.$scheme.options.issuedAt) || Date.now()
+    const _tokenTTL = getProp(result, this.$scheme.options.expiresIn) || this.$scheme.options.token.maxAge
+    const _tokenExpiresAt = getProp(result, this.$scheme.options.expiresAt) * 1000 || _tokenIssuedAt + (_tokenTTL * 1000)
+
+    try {
+      decodedToken = jwtDecode(token)
+      tokenExpiration = decodedToken.exp * 1000
+    } catch (error) {
+      // If we couldn't decode the token, use _tokenExpiresAt value
+      tokenExpiration = _tokenExpiresAt
```

Comment:
@MathiasCiarlo Ah maybe we could add it here. If `InvalidTokenError` then we set the `tokenExpiration` to `_tokenExpiresAt`. Otherwise throw error. Right?

Reply:
Yes! Absolutely!

Reply:
Fixed in commit https://github.com/nuxt-community/auth-module/pull/548/commits/20ed936aa50553187e1fa50372c7470c1b0fb5eb